### DISCUSSION TITLE
Making Biological assembly

### DISCUSSION THREAD

**Discussion by Aninda2020** [BradyAJohnston/MolecularNodes#248] [General]
I have imported a single chain of a multimeric protein from PDB. I wanted to ask how can I make protein assembly from this?

**Comment by BradyAJohnston** [comment]
You can use the `Biological Assembly` node from `Molecular Nodes -> Assembly -> Biological Assembly`. This node will instance all of the copies of the asymmetric unit that are encoded. 

This youtube video goes over the old version of the node. It is slightly different but works basically the same: https://www.youtube.com/watch?v=mq_8YCivVoY

**Comment by JuhaHuiskonen** [comment]
I tried importing PDB:1cd3 as in your tutorial.  When I add the "Biological assembly" node , I get the error "Unable to detect biological assembly information"

The full log is below.  I am using Blender 3.6.0 and Molecular Nodes 2.7.4

bpy.ops.object.delete(use_global=False)
bpy.context.scene.mol_pdb_code = "1cd3"
bpy.ops.mol.import_protein_rcsb()
Imported '1cd3' as 1cd3
bpy.context.area.ui_type = 'GeometryNodeTree'
bpy.ops.node.select(deselect_all=True, select_passthrough=True, location=(1602, 241))
bpy.ops.mol.assembly_bio()
Unable to detect biological assembly information.

**Reply by BradyAJohnston** [reply]
I have tested on Windows and Mac and I can't reproduce this error at all, mine works fine. Can you make sure that you have the protein selected and nothing else, when you try to make the biological assembly node?

**Comment by JuhaHuiskonen** [comment]
I hadn't realised the protein needs to be selected.  When it is, I can add the "Biological assembly" node without any error.  However, when I insert the node just before the output, the protein just disappears and I cannot get the full assembly. The output of MOL_assembly is "empty geometry". Any suggestions?

**Comment by hollypbird** [comment]
Is there a way to manually import assemblies from the pdb if they aren't showing up in Blender? Or does that mean the author didn't provide them? For example, 
[8to2.pdb1.gz](https://github.com/user-attachments/files/16663563/8to2.pdb1.gz)
I downloaded the assembly from the pdb website. 
<img width="1398" alt="Screenshot 2024-08-19 at 10 40 27 AM" src="https://github.com/user-attachments/assets/92c201db-be82-4e00-b8c5-086f2cda8600">
Thanks for the youtube tutorial on assembling a virus! I have managed to get an assembly to work in the past by following it.

**Reply by BradyAJohnston** [reply]
Hi @hollypbird, looking at the structure page on RCSB, there is no biological assembly for this structure. The structure that is included in the file is the whole structure.

**Reply by hollypbird** [reply]
Thanks so much, Brady for taking the time to look up the structure on RCSB!
